crates/standalone/src/energy_monitor.rs

```diff
@@ -48,15 +38,33 @@ impl EnergyMonitor for StandaloneEnergyMonitor {
                 .upgrade()
                 .expect("Worker env was dropped.")
         };
-        tokio::spawn(async move {
-            standalone_env
-                .withdraw_energy(&module_identity, energy_used.as_quanta())
-                .await
-                .unwrap();
-        });
+        tokio::spawn(async move { standalone_env.withdraw_energy(&identity, amount).await.unwrap() });
     }
 }
 
+impl EnergyMonitor for StandaloneEnergyMonitor {
+    fn reducer_budget(&self, _fingerprint: &ReducerFingerprint<'_>) -> ReducerBudget {
+        // Infinitely large reducer budget in Standalone
+        ReducerBudget::new(u64::MAX)
+    }
+
+    fn record_reducer(
+        &self,
+        fingerprint: &ReducerFingerprint<'_>,
+        energy_used: EnergyQuanta,
+        _execution_duration: Duration,
+    ) {
+        self.withdraw_energy(fingerprint.module_identity, energy_used)
+    }
+
+    fn record_disk_usage(&self, database: &Database, _instance_id: u64, disk_usage: u64, period: Duration) {
+        let amount = EnergyQuanta::from_disk_usage(disk_usage, period);
+        self.withdraw_energy(database.identity, amount)
+    }
```

Comment:
Note we need to be able to identify somewhere that this energy came from disk usage. See here for context. This could be recorded in `EnergyQuanta` itself or as an extra parameter to `withdraw_energy`.

Reply:
Right - I figured that's something that could be done as a follow-up; I wasn't sure how to do that since right now it seems the sole place energy usage is being stored is the controldb as just an integer of the total balance, and also since this is standalone it shouldn't really matter too much.

Reply:
I'm fine with that as long as this lands before @mamcx's.